plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -228,6 +233,55 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        int lengthLimit = toType.getLength().orElse(Integer.MAX_VALUE);
+        if (fromType.isShort()) {
+            return new ShortDecimalToVarcharCoercer(fromType, lengthLimit);
+        }
+        return new LongDecimalToVarcharCoercer(fromType, lengthLimit);
+    }
+
+    private static class ShortDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        private final int lengthLimit;
+
+        protected ShortDecimalToVarcharCoercer(DecimalType fromType, int lengthLimit)
+        {
+            super(fromType, VARCHAR);
+            this.lengthLimit = lengthLimit;
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            String stringValue = Decimals.toString(fromType.getLong(block, position), fromType.getScale());
+            // This may result in truncation fo digits before the decimal dot but this is what hive does
+            toType.writeString(blockBuilder, stringValue.substring(0, min(lengthLimit, stringValue.length())));
```

Comment:
cc @alexjo2144 @findinpath

Reply:
Let's turn this from user / analyst perspective.

Should the user know that the numbers added in the calculation have been truncated (even though Hive ignores this aspect) ? 
I would say strongly yes.


The use should be given a hint on how to cope with such a problem so that the column varchar type gets eventually a higher length/unbounded.

Reply:
> The use should be given a hint on how to cope with such a problem so that the column varchar type gets eventually a higher length/unbounded.

That's good idea, but not a must-have IMO.
If such cases are extremely rare, we can go without "nice hints".
We don't know yet how rare they are, so it's fine to add those hints lazily, when needed.